OpenRA.Game/Server/Server.cs

```diff
@@ -584,23 +584,28 @@ public void DropClient(Connection toDrop, int frame)
 				// Send disconnected order, even if still in the lobby
 				DispatchOrdersToClients(toDrop, 0, new ServerOrder("Disconnected", "").Serialize());
 
-				LobbyInfo.Clients.RemoveAll(c => c.Index == toDrop.PlayerIndex);
-
-				// Client was the server admin
-				// TODO: Reassign admin for game in progress via an order
-				if (Dedicated && dropClient.IsAdmin && State == ServerState.WaitingPlayers)
+				// Client was the server admin, but the game isn't over yet.
+				if (Dedicated && dropClient.IsAdmin)
 				{
-					// Remove any bots controlled by the admin
-					LobbyInfo.Clients.RemoveAll(c => c.Bot != null && c.BotControllerClientIndex == toDrop.PlayerIndex);
+					// Still in the lobby so handle things directly.
+					if (State == ServerState.WaitingPlayers)
+					{
+						// Remove the previous admin.
+						LobbyInfo.Clients.RemoveAll(c => c.Index == toDrop.PlayerIndex);
 
-					var nextAdmin = LobbyInfo.Clients.Where(c1 => c1.Bot == null)
-						.MinByOrDefault(c => c.Index);
+						// Remove any bots controlled by the previous admin.
+						LobbyInfo.Clients.RemoveAll(c => c.Bot != null && c.BotControllerClientIndex == toDrop.PlayerIndex);
 
-					if (nextAdmin != null)
-					{
-						nextAdmin.IsAdmin = true;
-						SendMessage("{0} is now the admin.".F(nextAdmin.Name));
+						// Find a new admin.
+						var nextAdmin = LobbyInfo.Clients.Where(c1 => c1.Bot == null).MinByOrDefault(c => c.Index);
+						if (nextAdmin != null)
+						{
+							nextAdmin.IsAdmin = true;
+							SendMessage("{0} is now the admin.".F(nextAdmin.Name));
+						}
 					}
+					else
+						DispatchOrdersToClients(toDrop, 0, new ServerOrder("NewAdmin", "").Serialize());
```

Comment:
Whenever I send this with a frame (frame + 1, frame - 1, frame + 2) whatever doesn't matter the game will crash, because the frame already contains an order (probably the drop/disconnect one).

![image](https://cloud.githubusercontent.com/assets/756669/14586584/8ea0f2cc-049d-11e6-8bb8-4f269da7d675.png)

 If I issue an order through `Game.OrderManager` the dedicated server will crash, as OrderManager is probably null there.

I have no idea how to solve this.